### PR TITLE
Introduce `fromImage` field in jib builder interface.

### PR DESCRIPTION
**Pull request by gsquared94**
**Related**: #4713 
This PR introduces `fromImage` as a property in the `jib` builder to override the configured base image. 
It's required in the implementation of [supporting artifact dependencies](https://github.com/GoogleContainerTools/skaffold/blob/master/docs/design_proposals/artifact-dependencies.md#jib-builder)

### CODE REVIEW

**Review by briandealwis — COMMENTED** [review]
LGTM, but let's get confirmation from the Jib team if they prefer `baseImage` or `fromImage`.

File: integration/examples/jib-sync/README.md [hunk 1]
```diff
@@ -63,34 +63,23 @@ build:
...
-    image = "gcr.io/distroless/java:debug"
-  }
-}
+```yaml
+build:
+  artifacts:
+  - image: skaffold-example
+    context: .
+    jib: 
+      baseImage: gcr.io/distroless/java:debug
```
[the remark this answers is not in the document]
Comment (chanseokoh) [reply integration/examples/jib-sync/README.md:77]: I like `fromImage`, as the Jib parameter is called `jib.from.image`.

[the remark this answers is not in the document]
Comment (gsquared94) [reply integration/examples/jib-sync/README.md]: changed

File: pkg/skaffold/schema/latest/config.go [hunk 2]
```diff
@@ -1077,4 +1077,7 @@ type JibArtifact struct {
 	// `maven`: for Maven.
 	// `gradle`: for Gradle.
 	Type string `yaml:"type,omitempty"`
+
+	// BaseImage overrides the default jib base image.
```
Comment (briandealwis) [inline pkg/skaffold/schema/latest/config.go:1081]: It's not just the default image (gcr.io/distroless/java:{8,11}).
```suggestion
	// BaseImage overrides the configured jib base image.
```

**Review by briandealwis — APPROVED** [review]

**Pull request merged**